pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -569,26 +597,76 @@ func (ssr *Snapshotter) snapshotEventHandler(stopCh <-chan struct{}) error {
 			if err != nil {
 				return err
 			}
+			if ssr.healthConfig.Enabled {
+				ctx, cancel := context.WithTimeout(context.TODO(), ssr.config.DeltaSnapshotPeriod.Duration)
+				latestFullSnapshot, latestDeltaSnapshotList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ssr.store)
+				if err == nil {
+					if err = heartbeat.UpdateDeltaSnapshotLease(ctx, ssr.logger, latestFullSnapshot, latestDeltaSnapshotList, ssr.k8sClientset, ssr.healthConfig.DeltaSnapshotLeaseName); err != nil {
+						ssr.logger.Warnf("1 Failed to update delta snapshot lease with error: %v", err)
+					}
+				} else {
+					ssr.logger.Warnf("failed to get latest snapshots from store with error: %v", err)
+				}
+				cancel()
+			}
 
 		case <-ssr.fullSnapshotTimer.C:
 			if _, err := ssr.TakeFullSnapshotAndResetTimer(); err != nil {
 				return err
 			}
+			if ssr.healthConfig.Enabled {
+				ctx, cancel := context.WithTimeout(context.TODO(), ssr.config.DeltaSnapshotPeriod.Duration)
+				if err := heartbeat.UpdateFullSnapshotLease(ctx, ssr.logger, ssr.PrevFullSnapshot, ssr.k8sClientset, ssr.healthConfig.FullSnapshotLeaseName); err != nil {
+					ssr.logger.Warnf("2 Failed to update full snapshot lease with error: %v", err)
```

Comment:
Why `2`?

Reply:
Sorry, that was used only for debugging
removed